js_modules/dagit/packages/core/src/asset-graph/__stories__/SidebarAssetInfo.stories.tsx

```diff
@@ -0,0 +1,261 @@
+        },
+        metadataEntries: [],
+        partitionDefinition: null,
+        assetKey: {
+          path: ['asset1'],
+          __typename: 'AssetKey',
+        },
+        op: {
+          name: 'asset1',
+          description: null,
+          metadata: [
+            {
+              key: 'compute_kind',
+              value: 'pandas',
+              __typename: 'MetadataItemDefinition',
+            },
+            {
+              key: 'kind',
+              value: 'pandas',
+              __typename: 'MetadataItemDefinition',
+            },
+          ],
+          __typename: 'SolidDefinition',
+        },
+        opVersion: null,
+        repository: MockRepo,
+        requiredResources: [],
+        type: {
+          key: 'Any',
+          name: 'Any',
+          displayName: 'Any',
+          description: null,
+          isNullable: false,
+          isList: false,
+          isBuiltin: true,
+          isNothing: false,
+          metadataEntries: [],
+          inputSchemaType: {
+            __typename: 'CompositeConfigType',
+            key: 'Selector.f2fe6dfdc60a1947a8f8e7cd377a012b47065bc4',
+            description: null,
+            isSelector: true,
+            typeParamKeys: [],
+            fields: [],
+            recursiveConfigTypes: [],
+          },
+          outputSchemaType: null,
+          __typename: 'RegularDagsterType',
+          innerTypes: [],
+        },
+      },
+    },
+  },
+};
+
+const EventsMock: MockedResponse<AssetEventsQuery> = {
+  request: {
+    query: ASSET_EVENTS_QUERY,
+    variables: {
+      assetKey: {path: ['asset1']},
+      before: undefined,
+      limit: 100,
+    },
+  },
+  result: {
+    data: {
+      __typename: 'DagitQuery',
+      assetOrError: {
+        __typename: 'Asset',
+        key: MockAssetKey,
+        id: '["asset1"]',
+        definition: {
+          __typename: 'AssetNode',
+          id: 'test.py.repo.["asset1"]',
+          partitionKeys: [],
+        },
+        assetMaterializations: [
+          {
+            __typename: 'MaterializationEvent',
+            description: '1234',
+            runId: '12345',
+            metadataEntries: [],
+            partition: null,
+            timestamp: '12345678654',
+            assetLineage: [],
+            label: null,
+            stepKey: 'op',
+            tags: [],
+            runOrError: {
+              __typename: 'Run',
+              pipelineName: '__ASSET_JOB_1',
+              mode: 'default',
+              pipelineSnapshotId: null,
+              id: '12345',
+              status: RunStatus.SUCCESS,
+              runId: '12345',
+              repositoryOrigin: {
+                __typename: 'RepositoryOrigin',
+                id: 'test.py',
+                repositoryLocationName: 'repo',
+                repositoryName: 'test.py',
+              },
+            },
+          },
+        ],
+        assetObservations: [
+          {
+            __typename: 'ObservationEvent',
```

Comment:
buildObservationEvent?

Reply:
Ahh man, I tried but I'm having trouble with the typescript unions for MetadataEntries conflicting. Will keep digging into that separately and see if i can figure out why, but it seems like the generators are returning instances of types that are not /quite/ what is specified by the fragment type somehow.